src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -246,5 +248,43 @@ private static int LocateFirstFoundByte(ulong match)
                                                0x03ul << 32 |
                                                0x02ul << 40 |
                                                0x01ul << 48) + 1;
+
+        public static bool TryGetGuid(ReadOnlySpan<byte> span, bool stringHasEscaping, out Guid value)
+        {
+            Debug.Assert(span.Length <= JsonConstants.MaximumEscapedGuidLength);
+
+            int bytesConsumed = 0;
+
+            value = default;
+
+            if (stringHasEscaping)
+            {
+                int idx = span.IndexOf(JsonConstants.BackSlash);
```

Comment:
nit: move the `stringHasEscaping` block to a helper method so that `TryGetGuid` might get inlined to the calling method for the common case.

Reply:
The non-escapsed path has been refactored into the public method, so we don't need inlining.
The escaped path has been moved to a helper.